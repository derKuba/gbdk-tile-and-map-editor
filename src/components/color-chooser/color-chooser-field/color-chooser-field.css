.gb-green {
  background-color: #6b790e;
}

.gb-grey {
  background-color: #c5bfbf;
}

.gb-grey-dark {
  background-color: #5a5760;
}

.gb-white {
  background-color: #fff;
}

.gb-black {
  background-color: #000;
}

.tile {
  width: 40px;
  height: 40px;
  flex: 1;
  margin: 0 10px;
  border: 1px solid #c0c0c0;
}

.tile:first-of-type {
  margin-left: 0;
}

.active {
  border: 5px solid #33ccff;
  width: 32px;
  height: 32px;
}
