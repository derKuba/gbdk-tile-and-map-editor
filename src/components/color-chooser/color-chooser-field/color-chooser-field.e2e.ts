import { newE2EPage } from '@stencil/core/testing';

describe('mvjp-color-chooser', () => {
    it('renders', async () => {
        const page = await newE2EPage();
        await page.setContent('<mvjp-color-chooser></mvjp-color-chooser>');

        const element = await page.find('mvjp-color-chooser');
        expect(element).toHaveClass('hydrated');
    });
});
