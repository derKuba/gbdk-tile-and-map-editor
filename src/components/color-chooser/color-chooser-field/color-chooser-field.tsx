import { Component, h, Prop } from '@stencil/core';

@Component({
    tag: 'mvjp-color-chooser-field',
    shadow: false,
    styleUrl: 'color-chooser-field.css',
})
export class ColorChooserField {
    /**
     *   choosen colorcode
     */
    @Prop() color: string;

    /**
     *   choosen field
     */
    @Prop() isActive: boolean;

    /**
     *   handle click
     */
    @Prop() clickTile: Function;

    render() {
        const colorClass = `gb-${this.color} ${this.isActive ? 'active' : ''}`;

        return <div class={`tile ${colorClass}`} onClick={() => this.clickTile(this.color)}></div>;
    }
}
