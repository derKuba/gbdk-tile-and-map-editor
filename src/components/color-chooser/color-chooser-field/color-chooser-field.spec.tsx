import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { ColorChooserField } from './color-chooser-field';

describe('mvjp-color-chooser', () => {
    it('builds', () => {
        expect(new ColorChooserField()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [ColorChooserField],
            template: () => {
                return (
                    <div>
                        <mvjp-color-chooser-field color={'white'} isActive={true} clickTile={() => {}} />
                        <mvjp-color-chooser-field color={'black'} isActive={false} clickTile={() => {}} />
                    </div>
                );
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('click on clickhandler', async () => {
        const clickMock = jest.fn();
        const page = await newSpecPage({
            components: [ColorChooserField],
            template: () => {
                return <mvjp-color-chooser-field color={'black'} isActive={false} clickTile={clickMock} />;
            },
        });

        page.root.querySelector('.tile').click();
        await page.waitForChanges();

        expect(clickMock).toBeCalledWith('black');
    });
});
