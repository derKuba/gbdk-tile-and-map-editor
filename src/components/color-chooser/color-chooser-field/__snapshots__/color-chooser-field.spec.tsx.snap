// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-color-chooser matches snapshot 1`] = `
<mvjp-color-chooser-field>
  <div class="active gb-white tile"></div>
</mvjp-color-chooser-field>
`;
